### ISSUE TITLE
possible memory leak

### ISSUE THREAD

**Issue by Sorte1** [j4k0xb/webcrack#4]
### Describe the bug

It seems that the javascript GC doesn't clean up all the memory so after deobfuscating a large file I was left with 1200 MB of memory that wouldn't be cleaned up.

I'm running node v18.16.0.

### Expected Behaviour

```js
import path from 'path';
import { webcrack } from 'webcrack';
import fs from 'fs';

(async () => {
  let source = fs.readFileSync(path.join(__dirname, 'raw-game.js'), 'utf8');
  let deobfuscated = await webcrack(source, {
    maxIterations: 5,
  });
  fs.writeFileSync(path.join(__dirname, 'deobfuscated.js'), deobfuscated.code);
})();
```

### Code

```JavaScript
too long for pastebin.

https://cdn.discordapp.com/attachments/822252846027374634/1097603453007712367/raw-game.js
```

### Logs

```Text
no logs
```

**Comment by j4k0xb** [comment]
Thanks for reporting, confirmed that the large AST isn't garbage collected because m.capture is used in global variables and still holds a reference to a previously matched node:

https://github.com/j4k0xb/webcrack/blob/e90ba5457ea3022ad250e515b872c5a544e952b8/src/deobfuscator/stringArray.ts#L39-L41

https://github.com/codemod-js/codemod/blob/7c4f86ffe4826b6151bd34cb655ee3de88e5b681/packages/matchers/src/matchers/capture.ts#L9-L11

There are around 60 of these across the whole lib so it will take some time to fix

**Comment by j4k0xb** [comment]
Fixed in cc898653ef6db5952f0f8c1b4a4a71e03b3fd887

Now the memory decreases back to 20-30 Mb after running
